midp/gfx.js

```diff
@@ -278,10 +268,27 @@ var currentlyFocusedTextEditor;
         converterFunc(rgbData, abgrData, width, height, 0, width);
 
         context.putImageData(ctxImageData, 0, 0);
-
-        setImageData(imageData, width, height, context);
     };
 
+    function setImageDataDimensions(imageData, w, h) {
+        imageData.klass.classInfo.getField("I.width.I").set(imageData, w);
+        imageData.context.canvas.width = w;
+
+        imageData.klass.classInfo.getField("I.height.I").set(imageData, h);
+        imageData.context.canvas.height = h;
+    }
+
+    Override["javax/microedition/lcdui/ImageData.<init>.()V"] = function() {
+        this.context = document.createElement("canvas").getContext("2d");
```

Comment:
Is overriding the constructors of ImageData necessary? Couldn't we keep the old behavior of creating the canvas in the ImageDataFactory natives?
I mean, we're creating the canvas once anyway, why creating it in the ImageData constructor?